### PR TITLE
Add prometheus collector

### PR DESCRIPTION
**Pull request by fgimenez**
Exposes metrics about ChartConfigs, including status.

### CODE REVIEW

**Comment by fgimenez** [comment]
This is how the metric looks like:
```
# HELP chart_operator_chartconfig_status Managed charts status.
# TYPE chart_operator_chartconfig_status gauge
chart_operator_chartconfig_status{channel_name="5-5-beta",chart_name="tb-chart",release_name="tb-release",release_status="DEPLOYED"} 1
```

**Comment by fgimenez** [comment]
I've also taken the opportunity to set `versionBundleVersion` value to chart-operator-resource chart in the tests to `0.3.0`, so that we test the current WIP version on CI instead of `v1`.

File: service/collector/chart_config.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package collector
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (rossf7) [inline service/collector/chart_config.go]: Nitpick but I think this should be `metav1` to match our usual convention.

File: service/collector/chart_config.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			channelNameLabel,
+			releaseNameLabel,
+			releaseStatusLabel,
+		},
+		nil,
+	)
+)
+
+func (c *Collector) collectChartConfigStatus(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics for vpcs")
```
Comment (rossf7) [inline service/collector/chart_config.go]: `vpcs` looks wrong here.

File: service/collector/chart_config.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		)
+	}
+	c.logger.Log("level", "debug", "message", "finished collecting metrics for ChartConfigs")
+}
+
+func (c *Collector) getChartConfigs() ([]*chartState, error) {
+	r, err := c.g8sClient.CoreV1alpha1().
+		ChartConfigs("giantswarm").
+		List(v1.ListOptions{})
+
```
Comment (rossf7) [inline service/collector/chart_config.go]: I think the line break could go here.

File: service/collector/collector.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+func (c *Collector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- chartConfigDesc
+}
+
+func (c *Collector) Collect(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics")
+
+	collectFuncs := []func(chan<- prometheus.Metric){
+		c.collectChartConfigStatus,
+	}
```
Comment (rossf7) [inline service/collector/collector.go]: Good to use the updated approach here with the WaitGroup, As we may need to add more collector funcs later.

**Pull request merged**